ksql-test-util/src/test/java/io/confluent/ksql/test/util/ZooKeeperEmbeddedTest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.test.util;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.apache.zookeeper.Watcher;
+import org.apache.zookeeper.Watcher.Event.KeeperState;
+import org.apache.zookeeper.ZooKeeper;
+import org.junit.Test;
+
+
+public class ZooKeeperEmbeddedTest {
+
+  /**
+   * Test is only valid if Jetty is on the class path:
+   */
+  @SuppressWarnings("unused")
+  private final org.eclipse.jetty.server.Connector ensureClassOnClassPath = null;
+  @SuppressWarnings("unused")
+  private final org.eclipse.jetty.servlet.ServletContextHandler ensureClassOnClassPath2 = null;
+
+  @Test
+  public void shouldSupportMultipleInstancesRunning() throws Exception {
+    // Given:
+    final ZooKeeperEmbedded first = new ZooKeeperEmbedded();
+
+    try {
+      // When:
+      final ZooKeeperEmbedded second = new ZooKeeperEmbedded();
+
+      // Then:
+      assertCanConnect(first, "first");
+      assertCanConnect(second, "second");
+
+      second.stop();
+    } finally {
+      first.stop();
+    }
+  }
+
+  private static void assertCanConnect(
+      final ZooKeeperEmbedded server,
+      final String name
+  ) {
+    final CountDownLatch connectionLatch = new CountDownLatch(1);
+
+    final Watcher watcher = event -> {
+      if (event.getState() == KeeperState.SyncConnected) {
+        connectionLatch.countDown();
+      }
+    };
+
+    ZooKeeper zooKeeper = null;
+
+    try {
+      final String connectString = server.connectString();
+      zooKeeper = new ZooKeeper(connectString, 2000, watcher);
+      final boolean success = connectionLatch.await(5, TimeUnit.SECONDS);
```

Comment:
I'm worried this might add a flaky test that doesn't really add extra functionality - anytime we have zookeeper in our tests i've seen timeouts on Jenkins. I'd be comfortable committing this without a test for it (or if you want to make sure that the setting isn't changed, we can test for this property).

Reply:
I hear you, but on the flip side such a simple test might help us get to the bottom of _why_ ZK is cause test instability. 

This test actually tests multiple ZK can be started. Just testing the property is set does not: if the ZK version changes then other things may stop multiple from being started.

I'd prefer to leave it in and see how it goes. We can always remove it.